src/backend/web/validation.js

```diff
@@ -43,11 +44,27 @@ function validateNewFeed() {
     const valid = ajv.validate(feedSchema, feedData);
 
     if (!valid) {
-      res.status(400).send();
+      return res.status(400).send();
     } else {
-      next();
+      return next();
     }
   };
 }
 
-module.exports.validateNewFeed = validateNewFeed;
+const validateParam = (req, res, next) => {
+  param('id').isLength({ min: 10, max: 10 })(req, res, (err) => {
+    console.log(err);
+    try {
+      validationResult(req).throw();
+      next();
+    } catch (e) {
+      console.log(e);
+      res.status(400).json({ message: 'ID Length is invalid' });
+    }
+  });
+};
+
```

Comment:
Can you return a middleware instead? Like below. You are on the right track
```
function validateNewFeed() {
  return function (req, res, next) {
    const feedData = req.body;
    const valid = ajv.validate(feedSchema, feedData);

    if (!valid) {
      res.status(400).send();
    } else {
      next();
    }
  };
}
```

Reply:
so in the route you can just call validatePostsIdParam()

Reply:
Thanks, I changed it back and the validation name also is changed to validatePostsIdParam.